### PR TITLE
python312Packages.pyside6: fix eval on linux

### LINKED ISSUE #356072: Evaluation failure: python3Packages.pyside6 (Linux-only)
## Steps To Reproduce

1. build `python3Packages.pyside6` on x86-64 or Aarch64 Linux at any Nixpkgs commit after 73cdc95de4ec02ad7e1f2e5a5fbfe05564aff0d0.

## Build log

<details>

<summary>Build Log</summary>

```
error:
       … while evaluating the attribute 'drvPath'
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/lib/customisation.nix:365:7:
          364|     in commonAttrs // {
          365|       drvPath = assert condition; drv.drvPath;
             |       ^
          366|       outPath = assert condition; drv.outPath;

       … while evaluating 'strict' to select 'drvPath' on it
         at /builtin/derivation.nix:1:552:
       … while calling the 'derivationStrict' builtin
         at /builtin/derivation.nix:1:208:
       … while evaluating derivation 'pyside6-6.8.0.2'
         whose name attribute is located at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/make-derivation.nix:336:7

       … while evaluating attribute 'buildInputs' of derivation 'pyside6-6.8.0.2'
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/make-derivation.nix:383:7:
          382|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          383|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          384|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       … in the condition of the assert statement
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/lib/customisation.nix:352:23:
          351|             drvPath = assert condition; drv.${outputName}.drvPath;
          352|             outPath = assert condition; drv.${outputName}.outPath;
             |                       ^
          353|           } //

       … while evaluating the attribute 'handled'
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/check-meta.nix:507:7:
          506|       # or, alternatively, just output a warning message.
          507|       handled =
             |       ^
          508|         (

       … from call site
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/check-meta.nix:511:13:
          510|           else if valid == "no" then (
          511|             handleEvalIssue { inherit meta attrs; } { inherit (validity) reason errormsg; }
             |             ^
          512|           )

       … while calling 'handleEvalIssue'
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/check-meta.nix:269:38:
          268|
          269|   handleEvalIssue = { meta, attrs }: { reason , errormsg ? "" }:
             |                                      ^
          270|     let

       … caused by explicit throw
         at /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/stdenv/generic/check-meta.nix:281:8:
          280|         else throw;
          281|     in handler msg;
             |        ^
          282|

       error: Package ‘apple-sdk-15.0’ in /nix/store/py1rfigbv569s09j94f1pd3lq00irqd8-source/pkgs/by-name/ap/apple-sdk/package.nix:115 is not available on the requested hostPlatform:
         hostPlatform.config = "x86_64-unknown-linux-gnu"
         package.meta.platforms = [
           "x86_64-darwin"
           "i686-darwin"
           "aarch64-darwin"
           "armv7a-darwin"
         ]
         package.meta.badPlatforms = [
           {
             cpu = {
               bits = 32;
             };
           }
         ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.
```

</details>

## Additional context

https://github.com/NixOS/nixpkgs/pull/354599#issuecomment-2477945684

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.7, NixOS, 24.11 (Velociraptor), 5b900e8`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.92.0-dev-pre20241103-b1a0e3c`
 - nixpkgs: `/nix/store/8b352i30b2x7z0lpf48gqsp13i1670vd-source`
## Notify maintainers

@gebner 

cc @emilazy as #354599 is the likely cause for this. I don't know anything about darwin builds so I'm not sure how to fix it without breaking eval on macOS.

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by pbsds**
zhf https://github.com/NixOS/nixpkgs/issues/352882
fixes https://github.com/NixOS/nixpkgs/issues/356072

~~i've yet to see the build complete, but~~ i'm now off to bed

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by emilazy — APPROVED** [review]
Agh, sorry about this! Maybe `darwinVersionInputs` should include the conditional itself. I am confident that this fix is correct but won’t override the draft without @K900 taking a look.

File: pkgs/development/python-modules/shiboken6/default.nix [hunk 1]
```diff
@@ -31,15 +31,18 @@ stdenv'.mkDerivation (finalAttrs: {
...
+  buildInputs =
+    [
+      llvmPackages.llvm
+      llvmPackages.libclang
+      python.pkgs.qt6.qtbase
+      python.pkgs.ninja
+      python.pkgs.packaging
+      python.pkgs.setuptools
+    ]
+    ++ lib.optional stdenv.hostPlatform.isDarwin [
```
Comment (K900) [inline pkgs/development/python-modules/shiboken6/default.nix:43]: Shouldn't be a list, right?

Comment (emilazy) [reply pkgs/development/python-modules/shiboken6/default.nix:43]: No longer confident the fix is correct 🫠

Comment (pbsds) [reply pkgs/development/python-modules/shiboken6/default.nix]: it's so annoying that `toString` doesn't care about nested lists. We should at some point make structured attrs default

**Pull request merged**